mindmeld/components/entity_recognizer.py

```diff
@@ -71,51 +72,55 @@ def _get_model_config(self, **kwargs):  # pylint: disable=arguments-differ
         )
         return super()._get_model_config(loaded_config, **kwargs)
 
-    def fit(self, queries=None, label_set=None, incremental_timestamp=None, **kwargs):
-        """Trains the entity recognition model using the provided training queries.
-
-        Args:
-            queries (list[ProcessedQuery]): The labeled queries to use as training data.
-            label_set (list, optional): A label set to load. If not specified, use the default.
-            incremental_timestamp (str, optional): The timestamp folder to cache models in.
-        """
+    def fit(self,
+            queries=None,
+            label_set=None,
+            incremental_timestamp=None,
+            load_cached=True,
+            **kwargs):
         logger.info(
             "Fitting entity recognizer: domain=%r, intent=%r", self.domain, self.intent
         )
-
         # create model with given params
         self._model_config = self._get_model_config(**kwargs)
         model = create_model(self._model_config)
 
-        if not label_set:
-            label_set = self._model_config.train_label_set
-            label_set = label_set if label_set else DEFAULT_TRAIN_SET_REGEX
+        if not queries:
+            if not label_set:
+                label_set = self._model_config.train_label_set
+                label_set = label_set if label_set else DEFAULT_TRAIN_SET_REGEX
+            queries = self._get_queries_from_label_set(label_set)
+        elif not isinstance(queries, ProcessedQueryList):
+            queries = ProcessedQueryList.from_in_memory_list(queries)
```

Comment:
I find this code block repeated across most `Classifier ` sub-classes. Can we create a helper function in the `Classifier` class that will return the queries based on `label_set` and optional `queries `?

Reply:
Great idea!  I'll see what that looks like.